src/OpenTelemetry.Extensions.Enrichment/README.md

```diff
@@ -6,6 +6,165 @@
 Contains OpenTelemetry .NET SDK telemetry enrichment framework
 which is used for enrichment of traces.
 
+## Introduction
+
+Telemetry enrichment attaches various types of information to traces.
+You can use the telemetry enrichment framework to attach any custom information
+that you would like to be present in all of your traces.
+
+With telemetry enrichment framework, you don't need to worry about attaching
+the information carefully to each telemetry object you touch.
+Instead, if you implement your enricher class, it  takes care of the details
+automatically. You simply register your class with the enrichment framework
+and the enrichment framework will make sure to call the `Enrich()` method of your
+class every time there is an `Activity` in your app.
+
 ## Traces
 
-TBD
+Currently this package supports trace enrichment only.
+
+### Steps to enable OpenTelemetry.Extensions.Enrichment
+
+You can view an example project using Enrichment at
+`/examples/enrichment/Examples.Enrichment`.
+
+### Step 1: Install package
+
+Download the `OpenTelemetry.Extensions.Enrichment` package:
+
+```shell
+dotnet add package OpenTelemetry.Extensions.Enrichment --prerelease
+```
+
+### Step 2: Create enricher class
+
+Create your custom enricher class that inherits from the `TraceEnricher` class
+and override the `public override void Enrich(TraceEnrichmentBag bag)` method:
```

Comment:
nit: Override the "virtual" 😄 

> and override the `public virtual void Enrich(TraceEnrichmentBag bag)` method

Interesting that `TraceEnricher` is `abstract` but "Enrich" method is `virtual`. Maybe we should have made it...

```csharp
public abstract class TraceEnricher : BaseEnricher<TraceEnrichmentBag>
{
    protected TraceEnricher()
    {
    }

    public abstract void Enrich(ref TraceEnrichmentBag traceEnrichmentBag);
}
```

🤷

Reply:
Updated. Actually, I was hesitating between doing this change here or in a separate PR. Now I have made it - I added the `in `keyword instead of `ref`